DemocracyClub/yournextrepresentative
# Feat/update party importer

This is the underpinning work for phase 2 of party pages on Who ([Trello](https://trello.com/c/gziXi9a7/3366-wcivf-party-pages-part-ii)) 

There's a few things going on here which are mostly detailed in ref #2050, with an additional chunk to save the EC's json return against a party.

Test 1: Party alternative name
- Run the migrations
- Run `parties_import_from_ec`
- Hit the YNR API locally `http://localhost:8000/api/next/parties/{{party_id}}` (I recommend PP53 because they have a registered alt. name)
- Check the alternative name field - it should have a value if you've chosen a party who has an alternative name, or null if you haven't

Test 2: JSON field
- You've already run the migration and the mgmt command, so simply check any party in your database. the `ec_data` field should be populated.

Test 3: Emblems
- Manually add an emblem for a party - I did so with SQL but you can use any means you like. 
```sql
-- example
insert into parties_partyemblem(created,modified,"default",ec_emblem_id,image,description,party_id)
values (CURRENT_TIMESTAMP,CURRENT_TIMESTAMP,false,1337,'emblems/PP52/48_Emblem_48.jpg','test emblem',958);
```
- Run `parties_import_from_ec`
- The emblem you manually added should now have active set to false


Test 4: Descriptions
- Manually add a description for a party - I did so with SQL but you can use any means you like. 
```sql
-- example
insert into parties_partydescription(created,modified,description,date_description_approved,party_id)
values (CURRENT_TIMESTAMP,CURRENT_TIMESTAMP,'test description',CURRENT_TIMESTAMP,958);
```
- Run `parties_import_from_ec`
- The description you manually added should now have active set to false


```[tasklist]
### PR Checklist

```
